File: backend/app/models/usuario.py
import enum

from sqlalchemy import Boolean, Enum, String
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import BaseModel


class RolUsuario(str, enum.Enum):
    ADMIN = "ADMIN"  # Acceso total al sistema
    OPERADOR = "OPERADOR"  # Gestión de clientes, contratos, pagos
    TECNICO = "TECNICO"  # Gestión de instalaciones y soporte técnico
    AUDITOR = "AUDITOR"  # Solo lectura de reportes y auditoría
    SOPORTE = "SOPORTE"  # Atención al cliente, consultas básicas


class Usuario(BaseModel):
    __tablename__ = "usuarios"

    email: Mapped[str] = mapped_column(String(255), unique=True, nullable=False, index=True)
    hashed_password: Mapped[str] = mapped_column(String(255), nullable=False)
    nombre_completo: Mapped[str] = mapped_column(String(255), nullable=False)
    rol: Mapped[RolUsuario] = mapped_column(
        Enum(RolUsuario), nullable=False, default=RolUsuario.OPERADOR
    )
    is_active: Mapped[bool] = mapped_column(Boolean, default=True, nullable=False)

File: frontend/src/types/usuario.ts
export type RolUsuario = "ADMIN" | "OPERADOR" | "TECNICO" | "AUDITOR" | "SOPORTE";

export interface Usuario {
  id: string;
  email: string;
  nombre_completo: string;
  rol: RolUsuario;
  is_active: boolean;
  created_at: string;
}

export interface UsuarioCreate {
  email: string;
  nombre_completo: string;
  rol: RolUsuario;
  password: string;
}

export interface UsuarioUpdate {
  email?: string;
  nombre_completo?: string;
  rol?: RolUsuario;
  is_active?: boolean;
}

export interface ChangePasswordRequest {
  current_password: string;
  new_password: string;
}
